include/matx/transforms/cub.h

```diff
@@ -1092,6 +1096,163 @@ inline void ExecSort(OutputTensor &a_out,
   size_t temp_storage_bytes = 0;
 };
 
+struct CustomArgMaxCmp
+{
+  template <typename T>
+  __MATX_DEVICE__ __MATX_HOST__ __MATX_INLINE__ T operator()(const T &a, const T &b) const {
+    return thrust::get<1>(a) < thrust::get<1>(b) ? b : a;
+  }
+};
+
+template <typename OutputTensor, typename TensorIndexType, typename InputOperator, CUBOperation_t op, typename CParams = EmptyParams_t>
+class matxCubTwoOutputPlan_t {
+  using T1 = typename InputOperator::value_type;
+
+public:
+  matxCubTwoOutputPlan_t(OutputTensor &a_out, TensorIndexType &aidx_out, const InputOperator &a, const CParams &cparams, const cudaStream_t stream = 0) :
+    cparams_(cparams)
+  {
+#ifdef __CUDACC__
+    MATX_NVTX_START("", matx::MATX_NVTX_LOG_INTERNAL)
+
+    if constexpr (op == CUB_OP_REDUCE_ARGMAX) {
```

Comment:
Not a big deal, but this should probably just be a regular function parameter to avoid making new types for something that's not in the fast path

Reply:
Fixed